### PR TITLE
upstream changes

### PR DESCRIPTION
**Pull request by jem-davies**
@gregfurman 

Please can you take a look? 

I had an issue with some missing func `TemplateData()` in ./public/bloblang/view.go - that was added on [this commit](https://github.com/redpanda-data/benthos/commit/2f82397309490a54b8a509cbd90182b528809d21) a couple of weeks ago, I have edited the commit manually to include here. Not sure if this is best way?

### CODE REVIEW

File: public/service/message.go [hunk 1]
```diff
@@ -465,6 +465,8 @@ func (m *Message) BloblangMutateFrom(blobl *bloblang.Executor, from *Message) (*
 //
 // This method allows mappings to perform windowed aggregations across message
 // batches.
+//
+// Deprecated: Use the much more efficient BloblangExecutor method instead.
```
Comment (jem-davies) [inline public/service/message.go:469]: This func being depreciated has caused linter errors across the repo...

File: public/service/message.go [hunk 2]
```diff
@@ -493,6 +495,8 @@ func (b MessageBatch) BloblangQuery(index int, blobl *bloblang.Executor) (*Messa
...
```
Comment (jem-davies) [inline public/service/message.go:499]: This func being depreciated has caused linter errors across the repo...

File: public/service/message.go [hunk 3]
```diff
@@ -536,6 +540,8 @@ func (b MessageBatch) BloblangQueryValue(index int, blobl *bloblang.Executor) (a
 // Note that using overlay means certain functions within the Bloblang mapping
 // will behave differently. In the root of the mapping the right-hand keywords
 // `root` and `this` refer to the same mutable root of the output document.
+//
+// Deprecated: Use the much more efficient BloblangExecutor method instead.
```
Comment (jem-davies) [inline public/service/message.go:544]: This func being depreciated has caused linter errors across the repo...

File: public/service/message.go (hunk 1 above)
Comment (jem-davies) [reply public/service/message.go:469]: Might have to go through and replace the functions with BloblangExecutor?

Comment (gregfurman) [reply public/service/message.go:469]: I think we should replace the deprecated function with the new one

Comment (jem-davies) [reply public/service/message.go:469]: removed the depreciated functions

File: public/service/message.go (hunk 2 above)
Comment (jem-davies) [reply public/service/message.go:499]: ☝️

File: public/service/message.go (hunk 3 above)
Comment (jem-davies) [reply public/service/message.go:544]: ☝️

**Review by gregfurman — COMMENTED** [review]
Looking good. Just 2 places to change naming. Am going to run some tests for this on my local before giving an approve.

**Comment by gregfurman** [comment]
LGTM @jem-davies -- feel free to merge when ready

**Pull request merged**